src/store/store.tsx

```diff
@@ -226,40 +226,62 @@ export function Store({ children }: { children: React.ReactNode }) {
     const loadUserBalances = async (address: string) => {
         try {
             if (userBalances.length === 0) {
-                const mobulaResponse = await fetch('/api/mobula/fetch-wallet-balance', {
-                    method: 'POST',
-                    headers: {
-                        'Content-Type': 'application/json',
-                    },
-                    body: JSON.stringify({
-                        address,
-                    }),
-                })
+                const chainIds = defaultChainDetails
+                    .filter((detail) => detail.mainnet)
+                    .filter((detail) => supportedMobulaChains.map((chain) => chain.chainId).includes(detail.chainId))
+                    .map((detail) => detail.chainId)
+                    .join(',')
+
+                let attempts = 0
```

Comment:
In the future I would encapsulate the retry logic in a dedicated place and reuse for everything. Maybe even put it in the sdk and reuse for both ui and the api

Reply:
hmm yes, dont see why it should be in the sdk but could just be a utils function. Will make backlog task